protocol/rpcconsumer/rpcconsumer_server.go

```diff
@@ -216,14 +216,23 @@ func (rpccs *RPCConsumerServer) sendRelayWithRetries(ctx context.Context, retrie
 			} else {
 				relayResult, err := relayProcessor.ProcessingResult()
 				if err == nil {
-					utils.LavaFormatInfo("[+] init relay succeeded", []utils.Attribute{{Key: "chainID", Value: rpccs.listenEndpoint.ChainID}, {Key: "APIInterface", Value: rpccs.listenEndpoint.ApiInterface}, {Key: "latestBlock", Value: relayResult.Reply.LatestBlock}, {Key: "provider address", Value: relayResult.ProviderInfo.ProviderAddress}}...)
+					utils.LavaFormatInfo("[+] init relay succeeded",
+						utils.LogAttr("GUID", ctx),
+						utils.LogAttr("chainID", rpccs.listenEndpoint.ChainID),
+						utils.LogAttr("APIInterface", rpccs.listenEndpoint.ApiInterface),
+						utils.LogAttr("latestBlock", relayResult.Reply.LatestBlock),
+						utils.LogAttr("provider address", relayResult.ProviderInfo.ProviderAddress),
+					)
 					rpccs.relaysMonitor.LogRelay()
 					success = true
 					// If this is the first time we send relays, we want to send all of them, instead of break on first successful relay
 					// That way, we populate the providers with the latest blocks with successful relays
 					if !initialRelays {
 						break
 					}
+
+					// On init relays, we abuse this for loop to send to send the initial relays, so we need to recreate the relay processor on successful relay
```

Comment:
this fix is a wring usage, relay processors are used to store the relays regardless of the success criteria, better to replace used providers (or remove the provider from the current usedProviders) with the existing processor

Reply:
Fix in 73530f5787d4eec243341631d11164f2f0b0a2ec